### PR TITLE
Add `Microsoft.ContainerRegistry` provider to the verify providers list

### LINKED ISSUE #557: Verify `Microsoft.ContainerRegistry` provider
We should verify this provider for all the different steps.

image

### PR DESCRIPTION
**Pull request by MicroFish91**
Fixes #557

### CODE REVIEW

**Review by nturinski — COMMENTED** [review]
I'd like to approach this differently so there's less room for error.

My two ideas are:

1. Have an array that contains all potential providers that are required by ACA and just always use that array whenever we do `VerifyProvidersStep`
2. Create a util function that returns `VerifyProvidersStep` that contains all of the providers that we need. It can be smart about it by checking the context and see if there are `newXXXXName` properties on it. but honestly, I think we don't lose much by just always verifying that the providers are registered.

File: src/constants.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export const managedEnvironmentsId = 'managedEnvironments';
 export const containerAppsId = 'containerApps';
 export const appProvider: string = 'Microsoft.App';
 export const webProvider: string = 'Microsoft.Web';
+export const registryProvider: string = 'Microsoft.ContainerRegistry';
```
Comment (nturinski) [inline src/constants.ts:14]: I think we should be good to move all these consts to `getVerifyProvidersStep`

File: src/utils/getVerifyProvidersStep.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+import { VerifyProvidersStep } from "@microsoft/vscode-azext-azureutils";
+import type { ISubscriptionActionContext } from "@microsoft/vscode-azext-utils";
+import { appProvider, operationalInsightsProvider, registryProvider, webProvider } from "../constants";
+
+/**
+ * Use to obtain a `VerifyProvidersStep` that registers all known container app providers to the user's subscription
+ */
+export function getVerifyProvidersStep<T extends ISubscriptionActionContext>(): VerifyProvidersStep<T> {
```
Comment (nturinski) [inline src/utils/getVerifyProvidersStep.ts:13]: I don't think you need to technically provide a context here based off the old way we used to create them.

File: src/constants.ts (hunk 1 above)
Comment (MicroFish91) [reply src/constants.ts:14]: We can move them there, but they won't be localized to that file so will still have to export a few of them

File: src/utils/getVerifyProvidersStep.ts (hunk 2 above)
Comment (MicroFish91) [reply src/utils/getVerifyProvidersStep.ts:13]: Yeah that makes sense, I'll just have to return using type inference instead

File: src/constants.ts (hunk 1 above)
Comment (nturinski) [reply src/constants.ts:14]: Oh are they? Never mind then-- I'm surprised we are using them anywhere else.

**Review by nturinski — APPROVED** [review]

File: src/constants.ts (hunk 1 above)
Comment (MicroFish91) [reply src/constants.ts:14]: two are, two aren't :P

**Pull request merged**